File: 2курс/2семестр/Код/ООП/2/Lab2/Lab2/noncomorg.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab2
{
    internal class NonComOrg : Organization
    {
        private string _purpose;
        private string _foundingSource;


        public string Purpose
        {
            get { return _purpose; }
            set { _purpose = value; }
        }
        public string FoundingSource
        {
            get { return _foundingSource; }
            set { _foundingSource = value; }
        }

        public NonComOrg(int id, string name, string inn, int employees, string purpose, string foundingSource)
            : base(id, name, inn, employees)
        {
            _purpose = purpose;
            _foundingSource = foundingSource;
        }
        public NonComOrg(string name, string inn, int employees, string purpose, string foundingSource)
            : base(name, inn, employees)
        {
            _purpose = purpose;
            _foundingSource = foundingSource;
        }

        public NonComOrg() : base("Фонд Помощи", "0987654321", 10)
        {
            _purpose = "Помощь нуждающимся";
            _foundingSource = "Гранты";
        }

        public override string Report()
        {
            return $"Некоммерческая организация: {Name}\n" +
                   $"Цель: {_purpose}\n" +
                   $"Источник финансирования: {_foundingSource}";
        }
        public override int Taxes()
        {
            return Employees * 10;
        }
        public string Program(string program)
        {
            return program;
        }
        public string AttractFunding(string source)
        {
            return source;
        }

    } 
}

File: 2курс/2семестр/Код/ООП/3/laba3/src/main/java/ru/university/lab3/entity/NonComOrg.java
package ru.university.lab3.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "noncomorganizations")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
public class NonComOrg extends Organization {
    private String purpose;
    private String source;

    public NonComOrg(String name, String inn, int employeesCount, String purpose, String source) {
        this.setName(name);
        this.setInn(inn);
        this.setEmployeesCount(employeesCount);
        this.purpose = purpose;
        this.source = source;
    }

    @Override
    public String report() {
        return "Некоммерческая организация: " + getName()
                + " | ИНН: " + getInn()
                + " | Сотрудники(по штату): " + getEmployeesCount()
                + " | Цель: " + getPurpose()
                + " | Источники: " + getSource();
    }

    @Override
    public int taxes() {
        return getEmployeesCount() * 10;
    }

    public String program(String programName) {
        System.out.println("Запускаем программу: " + programName);
        return programName;
    }

    public String attractFunding(String newSource) {
        System.out.println("Привлекаем финансирование от: " + newSource);
        this.setSource(this.getSource() + ", " + newSource);
        return getSource();
    }
}